src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -1258,21 +1258,93 @@ private void AnalyzeScenarios(int[] scenarios)
         }
 
         [Fact]
-        public void AnalyzeCommandBase_MultithreadedShouldUseCacheIfFilesAreTheSame()
+        public void AnalyzeCommandBase_Multithreaded_ShouldOnlyLogArtifactsWhenHashesIsEnabled()
```

Comment:
>AnalyzeCommandBase_Multithreaded_ShouldOnlyLogArtifactsWhenHashesIsEnabled [](http://example.com/codeflow?start=20&length=74)

This test will guarantee that we are always producing the correct number of artifacts if hashes is enabled for the multithreaded analysis.

If not enabled, the artifacts should be null.